HuliWebShop/src/main/java/huli/example/huliwebshop/controllers/CategoryController.java

```diff
@@ -1,19 +1,47 @@
 package huli.example.huliwebshop.controllers;
 
+import huli.example.huliwebshop.DTOs.ProductCreateDTO;
+import huli.example.huliwebshop.models.Category;
 import huli.example.huliwebshop.services.CategoryService;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.CrossOrigin;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
 @CrossOrigin(origins = "*")
 @RestController
+@RequestMapping("/api/categories")
 public class CategoryController {
     CategoryService categoryService;
 
     @Autowired
     public CategoryController(CategoryService categoryService) {
         this.categoryService = categoryService;
     }
+    @PostMapping("/admin/create")
+    public ResponseEntity createCategory(@RequestBody Category category){
+        try {
+            return ResponseEntity.ok().body(categoryService.saveNewCategory(category));
+        } catch(Exception e){
+            return ResponseEntity.status(400).body(e.getMessage());
```

Comment:
It's connected to the other one, but causing another issue: generally catching every exception and labeling it as a bad request (http400) is not correct, because other things can happen as well that is not a client side error (like the database is temp unavailable or a bug)